d2bs/kolbot/libs/GameData.js

```diff
@@ -4,57 +4,173 @@
+const AREALOCALESTR = [5389, 5055, 5054, 5053, 5052, 5051, 5050, 5049, 5048, 5047, 5046, 5045, 5044, 5043, 5042, 5041, 5040, 5039, 5038, 5037, 5036, 5035, 5034, 5033, 5032, 5031, 5030, 5029, 5028, 5027, 5026, 5025, 5024, 5023, 5022, 5021, 5020, 5019, 5018, 788, 852, 851, 850, 849, 848, 847, 846, 845, 844, 843, 842, 841, 840, 839, 838, 837, 836, 835, 834, 833, 832, 831, 830, 829, 828, 827, 826, 826, 826, 826, 826, 826, 826, 825, 824, 820, 819, 818, 817, 816, 815, 814, 813, 812, 810, 811, 809, 808, 806, 805, 807, 804, 845, 844, 803, 802, 801, 800, 799, 798, 797, 796, 795, 790, 792, 793, 794, 791, 789, 22646, 22647, 22648, 22649, 22650, 22651, 22652, 22653, 22654, 22655, 22656, 22657, 22658, 22659, 22660, 22662, 21865, 21866, 21867, 22663, 22664, 22665, 22667, 22666, 5389, 5389, 5389, 5018];
+const MONSTERKEYS = [
+	['mon1', 'mon2', 'mon3', 'mon4', 'mon5', 'mon6', 'mon7', 'mon8', 'mon9', 'mon10'],
+	['nmon1', 'nmon2', 'nmon3', 'nmon4', 'nmon5', 'nmon6', 'nmon7', 'nmon8', 'nmon9', 'nmon10'],
+][me.diff && 1]; // mon is for normal, nmon is for nm/hell, umon is specific to picking champion/uniques in normal
 
-include ('AreaData.js');
-include ('MonsterData.js');
-include ('ExpData.js');
+/**
+ *  MonsterData[classID]
+ *  .Index = Index of this monster
+ *  .Level = Level of this monster in normal (use GameData.monsterLevel to find monster levels)
+ *  .Ranged = if monster is ranged
+ *  .Rarity = weight of this monster in level generation
+ *  .Threat = threat level used by mercs
+ *  .Align = alignment of unit (determines what it will attack)
+ *  .Melee = if monster is melee
+ *  .NPC = if unit is NPC
+ *  .Demon = if monster is demon
+ *  .Flying = if monster is flying
+ *  .Boss = if monster is a boss
+ *  .ActBoss = if monster is act boss
+ *  .Killable = if monster can be killed
+ *  .Convertable = if monster is affected by convert or mind blast
+ *  .NeverCount = if not counted as a minion
+ *  .DeathDamage = explodes on death
+ *  .Regeneration = hp regeneration
+ *  .LocaleString = locale string index for getLocaleString
+ *  .ExperienceModifier = percent of base monster exp this unit rewards when killed
+ *  .Undead = 2 if greater undead, 1 if lesser undead, 0 if neither
+ *  .Drain = drain effectiveness percent
+ *  .Block = block percent
+ *  .Physical = physical resist
+ *  .Magic = magic resist
+ *  .Fire = fire resist
+ *  .Lightning = lightning resist
+ *  .Poison = poison resist
+ *  .Minions = array of minions that can spawn with this unit
+ */
+
+var MonsterData = Array(MONSTERINDEXCOUNT);
+
+for (let i = 0; i < MonsterData.length; i++) {
+	let index = i;
```

Comment:
why do we need a copy of i? why not just use i?

Reply:
it's a scoping nuance, index is scoped at each iteration, and i isn't
if I don't do this then each new object I create refers to the last value i had instead of the value of index set for this iteration